torch/csrc/jit/frontend/function_schema_parser.cpp

```diff
@@ -27,7 +27,7 @@ namespace jit {
 namespace {
 struct SchemaParser {
   SchemaParser(const std::string& str)
-      : L(std::make_shared<Source>(str)),
+      : L(std::make_shared<SourceView>(c10::string_view(str))),
```

Comment:
why is it necessary to heap-allocate and reference-count the SourceView instead of storing it by value?